### DISCUSSION TITLE
Road-Rail Vehicle Simulation in Carla

### DISCUSSION THREAD

**Discussion by LeonTimmer99** [carla-simulator/carla#8306] [Q&A]
Hello,
I want to simulate a autonomous road-rail vehicle. Is this possible in carla? I want to have it drive on the road and on train tracks. How can train tracks be built in carla and how can I assign a vehicle to the tracks so it follows its path like a real train? Ideally I would like to let it switch from road to rail while in motion.

**Comment by Homa333** [comment]
Hey @LeonTimmer99 , were you able to make any progress simulating rail road in Carla.

**Comment by Chinese-Coding** [comment]
That is a good idea. I met this demand too. Is there any progress?
